src/riak_ql_parser.yrl

```diff
@@ -157,7 +159,16 @@ Endsymbol '$end'.
 Statement -> StatementWithoutSemicolon : '$1'.
 Statement -> StatementWithoutSemicolon semicolon : '$1'.
 
-GroupBy -> group by Fields: {group_by, '$3'}.
+GroupBy -> group by GroupByClause: {group_by, '$3'}.
+
+GroupByClause -> TimeFunc comma GroupByClause   : ['$1']++'$3'.
+GroupByClause -> Identifier comma GroupByClause : ['$1']++'$3'.
+GroupByClause -> asterisk   : return_error_flat("GROUP BY can only contain table columns and the time/2 function but '*' was found.").
```

Comment:
Creepily, stealthily migrating hand-crafted error messages to parser declarations. :+1: